src/components/Calculator/index.tsx

```diff
@@ -0,0 +1,76 @@
+import { useRef, useState } from "react";
+
+import {
+  CalOperatorTypes,
+  MAX_NUM_OF_NUMBERS,
+  MAX_NUMBER_LENGTH,
+  OperatorTypes,
+} from "@/util/constants";
+import { doOperate } from "@/util/helper";
+import messages from "@/util/messages";
+
+import DigitButtons from "./DigitButtons";
+import styles from "./index.module.css";
+import ModifierButton from "./ModifierButton";
+import OperatorButtons from "./OperatorButtons";
+import ResultPanel from "./ResultPanel";
+
+// eslint-disable-next-line no-useless-escape
+const operatorRegex = /[\+-\/X]/g;
+
+export default function Calculator() {
+  const [resultPanel, setResultPanel] = useState("");
+
+  const curNumber = useRef("");
+
+  const inputNumbers = resultPanel.split(operatorRegex);
+  const curOperator = resultPanel.match(operatorRegex) as [CalOperatorTypes];
+
+  function handleDigitButtonClick(buttonValue: number) {
+    if (curNumber.current.length >= MAX_NUMBER_LENGTH) {
+      alert(messages.OVER_MAX_NUMBER_LENGTH);
+      return;
+    }
+
+    curNumber.current += `${buttonValue}`;
+    setResultPanel((prev) => `${prev}${buttonValue}`);
+  }
+
+  function handleOperatorButtonClick(buttonValue: OperatorTypes) {
+    if (buttonValue !== "=" && resultPanel === "") {
+      alert(messages.CAL_WITH_EMPTY_NUMBER);
+      return;
+    }
+
+    if (buttonValue === "=") {
+      if (!curOperator) return;
+
+      const [num1, num2] = inputNumbers.map((number) => Number(number));
```

Comment:
```suggestion
      const [num1, num2] = inputNumbers.map(Number);
```

이런 식으로도 작성해 볼 수 있을 것 같습니다 :)